enclave_build/src/cache.rs

```diff
@@ -0,0 +1,219 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+#![allow(dead_code)]
+use std::{
+    collections::HashMap,
+    fs::{self, File, OpenOptions},
+    path::{Path, PathBuf},
+};
+
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+use crate::{EnclaveBuildError, Result};
+
+/// Root folder for the cache.
+pub const CACHE_ROOT_FOLDER: &str = "XDG_DATA_HOME";
+/// Path to the blobs folder
+pub const CACHE_BLOBS_FOLDER: &str = "blobs/sha256/";
+/// Name of the cache index file which stores the (image URI <-> image hash) mappings.
+pub const CACHE_INDEX_FILE_NAME: &str = "index.json";
+/// The name of the OCI layout file from the cache.
+pub const CACHE_OCI_LAYOUT_FILE: &str = "oci-layout";
+
+/// Constants used for complying with the OCI cache structure
+pub const DEFAULT_MEDIA_TYPE: &str = "application/vnd.oci.image.manifest.v1+json";
+pub const REF_ANNOTATION: &str = "org.opencontainers.image.ref.name";
+pub const OCI_LAYOUT: (&str, &str) = ("imageLayoutVersion", "1.0.0");
+
+/// Manifest entry in index file
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct ManifestEntry {
```

Comment:
I tried to match the `linuxkit` format of the index file, but I see that `oci-distirbution` has a more general structure which might work. This could also loosen the coupling with `linuxkit`.